byterun/win32.c

```diff
@@ -914,3 +918,58 @@ void caml_restore_win32_terminal(void)
   if (startup_codepage != 0)
     SetConsoleOutputCP(startup_codepage);
 }
+
+/* Detect if a named pipe corresponds to MSYS/Cygwin tty: see
+
+   https://github.com/mirror/newlib-cygwin/blob/master/winsup/cygwin/dtable.cc#L932
+*/
+typedef
+BOOL (WINAPI *tGetFileInformationByHandleEx)(
+  HANDLE                    hFile,
+  FILE_INFO_BY_HANDLE_CLASS FileInformationClass,
+  LPVOID                    lpFileInformation,
+  DWORD                     dwBufferSize
+);
+
+static int caml_win32_ismsystty(int fd)
+{
```

Comment:
Looks like this function could (should?) take a HANDLE as argument and not a file descriptor, saving a `_getosfhandle` call that is redundant with the one done in `caml_win32_isatty`.

Reply:
Indeed, well spotted!